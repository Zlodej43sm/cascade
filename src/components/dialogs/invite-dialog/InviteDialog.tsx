import React from 'react';

import DialogWrapper, { DialogWrapperProps } from 'lib/components/dialog-wrapper/DialogWrapper';
import InputField from 'lib/components/forms/input-field/InputField';
import TextAreaField from 'lib/components/forms/text-area-field/TextAreaField';
import Button, { ButtonSizes } from 'lib/components/buttons/button/Button';

import styles from './InviteDialog.scss';

enum FieldProps {
  EmailName = 'email',
  EmailInputLabel = 'Email invite to',
  EmailInputPlaceholder = 'Please enter valid email',
  MessageName = 'message',
  MessageTextareaLabel = 'Message',
  MessageTextareaPlaceholder = 'Can you please take the lead on these?',
}

export type InviteDialogProps = DialogWrapperProps & {
  onSubmit?: (event: React.MouseEvent<HTMLButtonElement>) => void
};

function InviteDialog({ onSubmit, ...wrapperProps }: InviteDialogProps) {
  return (
    <DialogWrapper title="Invite your Team" width={568} {...wrapperProps}>
      <div className={styles.container}>
        <InputField
          name={FieldProps.EmailName}
          label={FieldProps.EmailInputLabel}
          placeholder={FieldProps.EmailInputPlaceholder}
        />
        <TextAreaField
          name={FieldProps.MessageName}
          label={FieldProps.MessageTextareaLabel}
          placeholder={FieldProps.MessageTextareaPlaceholder}
        />
      </div>
      <footer className={styles.footer}>
        <Button onClick={onSubmit} size={ButtonSizes.Compact}>Invite</Button>
      </footer>
    </DialogWrapper>
  );
}

export default InviteDialog;
