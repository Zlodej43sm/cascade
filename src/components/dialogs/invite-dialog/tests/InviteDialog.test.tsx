import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import { inviteDialogArguments } from '../__mocks__/InviteDialog.mock';
import InviteDialog from '../InviteDialog';

describe('InviteDialog', () => {
  it('Should match snapshot', () => {
    const { asFragment } = render(<InviteDialog {...inviteDialogArguments} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should call apply handler', () => {
    const onSubmit = jest.fn();
    const { getByTestId } = render(<InviteDialog {...inviteDialogArguments} onSubmit={onSubmit} />);
    fireEvent.click(getByTestId('button'));
    expect(onSubmit).toHaveBeenCalled();
  });
});
