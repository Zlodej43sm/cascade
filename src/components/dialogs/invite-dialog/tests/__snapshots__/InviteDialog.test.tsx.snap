// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteDialog Should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="container"
    data-testid="dialog-wrapper-container"
    style="width: 568px;"
  >
    <header
      class="header"
    >
      <div
        class="title"
        data-testid="dialog-wrapper-title"
      >
        Invite your Team
      </div>
      <div
        class="closeButton"
      >
        <button
          aria-label="Close"
          class="button"
          data-testid="close-button"
          type="button"
        />
      </div>
    </header>
    <div
      class="content"
    >
      <div
        class="container"
      >
        <label
          class="wrapper"
        >
          <span
            class="label"
          >
            Email invite to
          </span>
          <input
            class="input"
            name="email"
            placeholder="Please enter valid email"
          />
        </label>
        <label
          class="wrapper"
        >
          <span
            class="label"
          >
            Message
          </span>
          <textarea
            class="input"
            name="message"
            placeholder="Can you please take the lead on these?"
          />
        </label>
      </div>
      <footer
        class="footer"
      >
        <button
          class="button primary compact"
          data-testid="button"
          type="button"
        >
          Invite
        </button>
      </footer>
    </div>
  </section>
</DocumentFragment>
`;
